engine/storage/snapshot/src/main/java/org/apache/cloudstack/storage/snapshot/SnapshotObject.java

```diff
@@ -127,7 +127,7 @@ public SnapshotInfo getChild() {
         if (vo == null) {
             return null;
         }
-        return snapshotFactory.getSnapshot(vo.getId(), store);
+        return snapshotFactory.getSnapshot(vo.getSnapshotId(), store);
```

Comment:
@Pearl1594 
this looks like a critical issue

Reply:
Actually Yes , however, it's usage comes into play only cleanup logic :
1. where in it decides whether to cleanup the snapshot on the secondary store or not 
2. When there are snapshots in destroyed state and the garbage collector kicks in, so the impact of this hasn't been big.